src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -229,9 +229,30 @@ internal static IntPtr GetDelegateFunctionPointer()
             Debug.Assert(fnPtr != null);
             return fnPtr;
         }
+        
+        private static bool? s_isTLSSupported;  
 
+        private static void EnsureTLSSupported()
```

Comment:
This is not necessary. We have to have some kind of TLS support to work.

`HAVE_THREAD_LOCAL` means what kind of TLS support is available, not whether it is available at all.